opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -123,6 +184,36 @@ public function index() {
         $this->response->setOutput($this->render());
     }
 
+    protected function handleCustomerSync() {
+        if (!$this->user->hasPermission('modify', 'module/smaily_for_opencart')) {
+            return;
+        }
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Declare customer sync objects.
+        $customer_sync_enabled = $this->request->post['smaily_for_opencart_sync_enabled'];
+        // For if no customer sync fields are selected.
+        $customer_sync_fields = array();
+        if (!empty($this->request->post['smaily_for_opencart_sync_fields']) && is_array($this->request->post['smaily_for_opencart_sync_fields'])) {
+            $customer_sync_fields = $this->request->post['smaily_for_opencart_sync_fields'];
+        }
+        $customer_sync_token = $this->request->post['smaily_for_opencart_sync_token'];
```

Comment:
Should I just remove anything not alphanumeric and save it to db, or disallow saving and display an error message?

Reply:
That is up to you. From the user's perspective it would be better if error is shown - creates less confusion.

3rd option would be to not allow user to set custom token, but rather generate it once and have a mechanism in place for user to regenerate the token if they should desire it.